### PR TITLE
Use log4cplus.properties In Client Service

### PR DESCRIPTION
**Pull request by irisk29**
Added watch thread for the client service logger.
The path for the log4cplus.properties is defined by an environment variable.

### CODE REVIEW

File: client/clientservice/src/main.cpp [hunk 1]
```diff
@@ -30,6 +30,13 @@ using concord::client::concordclient::ConcordClientConfig;
 
 namespace po = boost::program_options;
 
+const static int kLogConfigRefreshIntervalInMs = 60 * 1000;
+
+const static char* GetLog4CplusConfigLocation() {
```
Comment (robem) [inline client/clientservice/src/main.cpp:35]: Let's make this more user-friendly. If the environment variable isn't set then let's print a warning to `std::cerr`. Also, function names start with lower-case.

File: client/clientservice/src/main.cpp [hunk 2]
```diff
@@ -51,9 +58,8 @@ po::variables_map parseCmdLine(int argc, char** argv) {
 }
 
 int main(int argc, char** argv) {
-  // TODO: Use config file and watch thread for logger
   auto logger = logging::getLogger("concord.client.clientservice.main");
-
+  LOG_CONFIGURE_AND_WATCH(GetLog4CplusConfigLocation(), kLogConfigRefreshIntervalInMs);
```
Comment (robem) [inline client/clientservice/src/main.cpp:62]: We should configure the logger *before* we get it.

**Review by robem — CHANGES_REQUESTED** [review]

File: client/clientservice/src/main.cpp (hunk 1 above)
Comment (irisk29) [reply client/clientservice/src/main.cpp]: Done.

File: client/clientservice/src/main.cpp (hunk 2 above)
Comment (irisk29) [reply client/clientservice/src/main.cpp]: Fixed.

File: client/clientservice/src/main.cpp [hunk 3]
```diff
@@ -72,6 +78,7 @@ int main(int argc, char** argv) {
   auto metrics = std::make_shared<concordMetrics::Aggregator>();
   concord_client->setMetricsAggregator(metrics);
   ClientService service(std::move(concord_client));
+  LOG_CONFIGURE_AND_WATCH(getLog4CplusConfigLocation(), kLogConfigRefreshIntervalInMs);
```
Comment (robem) [inline client/clientservice/src/main.cpp:81]: We are retrieving the logger in line 62 and we configure log4cplus in line 81. Shouldn't we configure it before we get the logger or does it not matter? In any case, those two lines should be close together.

Comment (irisk29) [reply client/clientservice/src/main.cpp]: I checked - it works either way. I moved it to be before line 62.

**Review by robem — APPROVED** [review]

**Pull request merged**